cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/sidecar/GcpMetadataHttpHandlerInternal.java

```diff
@@ -123,29 +138,77 @@ public void token(HttpRequest request, HttpResponder responder,
               METADATA_FLAVOR_HEADER_KEY, METADATA_FLAVOR_HEADER_VALUE));
     }
 
-    // TODO: CDAP-20750
-    if (metadataServiceEndpoint == null) {
+    if (gcpMetadataTaskContext != null) {
```

Comment:
Should we consider throwing an error here if the metadata context is null? That likely indicates we didn't set something properly.

Reply:
Documented the edge case when it is required:
```
while initializing io.cdap.cdap.common.guice.DFSLocationModule$LocationFactoryProvider#get
in io.cdap.cdap.internal.app.worker.TaskWorkerTwillRunnable.
```